### ISSUE TITLE
NoMethodFound error when trying to add to table with curl requests

### ISSUE THREAD

**Issue by d00medman** [ga-wdi-boston/full-stack-project#815]
I have my relationships up and built (I am using movies as my second table), and am now working on tying the current_user to my movie.

In order to do this, I basically hard copied the code from examples and retrofitted it for movies:

`@movie = current_user.movies.build(movie_params)`

I created a user with curl and signed said user in with no friction. However, when I attempt to run the following curl script:

`curl --include --request POST http://localhost:4741/movies \
 --header "Content-Type: application/json" \
 --data '{
   "movie": {
     "title": "Mad Max: Fury Road"
   }
 }'`

I am thrown a 500 error telling me that the method "movies" is undefined. I am unsure of why this would be failing, as it is functionally a hard copy of the examples template. 

I am currently searching the issue queue for similar examples, but would appreciate any guidance. Frankly speaking, I am totally lost as to why this error would be throwing.

**Comment by jordanallain** [comment]
can you look at the rails log to see what the error was and paste it here?

**Comment by d00medman** [comment]
`"status":500,"error":"Internal Server Error","exception":"#\u003cNoMethodError: undefined method 'movies' for nil:NilClass\u003e"`

I switched the superclass in movies_controller to OpenReadController, to match that of the examples_controller, but it threw the following error when I attempted to create a movie via curl request:

`HTTP/1.1 401 Unauthorized
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
WWW-Authenticate: Token realm="Application"
Content-Type: text/plain; charset=utf-8
Cache-Control: no-cache
X-Request-Id: 52de6aa6-f309-44d9-8a22-1952ab12561a
X-Runtime: 0.255684
Vary: Origin
Transfer-Encoding: chunked'`

Though I think this is because I failed to include any token authentication in my curl script (I am not entirely sure how I would go about finding the appropriate token to include though).

**Comment by jordanallain** [comment]
when you sign in with a curl script the token will be returned to you in the terminal if successful

**Comment by d00medman** [comment]
Alright, I've found my token, I just can't figure out how to integrate it into a curl script. The command line tends to fire before I have the chance to finish typing it out.

How can I run the following curl script from the command line 

`API="${API_ORIGIN:-http://localhost:4741}"
URL_PATH="/movies"
curl "${API}${URL_PATH}" \
  --include \
  --request POST \
  --header "Content-Type: application/json" \
  --header "Authorization: Token token=$TOKEN" \
  --data '{
    "movie": {
      "title": "'"${TEXT}"'"
    }
  }'
echo`

**Comment by d00medman** [comment]
Also, was I on the right track in changing the superclass from an ApplicationController to an OpenReadController?

This change allowed me to bypass the 500 error in exchange for the more manageable (if more obnoxious) 400 errors, but I am not sure why this has happened or if it will FUBAR my API in the long run.

**Comment by jordanallain** [comment]
its up to you whether you want your movies controller to inherit from open read or protected, but i would probably choose one of those.

**Comment by jordanallain** [comment]
you can run curl scripts directly from the command line or you can save them in a script and run the script itself.

**Comment by d00medman** [comment]
I understand that. I suppose what I'm asking is how I am supposed to interpolate the token at the command line

`TEXT="MAD MAX" sh create-movie.sh` 

throws a 401 because I it is not given a proper token, but I am unsure of how to give it the proper token in this format.

**Comment by jordanallain** [comment]
you can either set a bash variable or just paste the token right in there

**Comment by d00medman** [comment]
It worked. First, I ran

`EMAIL="oog@chaka.com" PASSWORD="a" sh sign-in.sh'`

To sign into the API and create a token. Taking the token, I ran

`TEXT="MAD MAX" TOKEN="BAhJIiU2NzlmMjZhNDFiMThlNGZkMmU2ZDFlNmNlN2IyZTM1NAY6BkVG--0d2d68254142ba62c09bf782cad62a8daa16e5b3" sh create-movie.sh'`

I got a 201 success code in return, my SQL query returned that I had created MAD MAX with a user_id 0f 8 (the id of oog@chaka.com) and it is manifested on http://localhost:4741/movies/.

I think this means I'm good, but I would appreciate it if there is anything flawed in my methodology which could be fixed now to save me a headache later.
